### PR TITLE
test(hts221): Add level-1 hardware unit test.

### LINKED ISSUE #122: test(hts221): Add hardware-unit test (level 1) using shared driver_checks.h
## Context

The repo has a rigorous native-mock test suite for HTS221 (`tests/native/test_hts221/`, 14 tests), but zero hardware validation beyond the smoke-test sketch. This issue creates the reference **level-1 hardware-unit** test suite for HTS221: same contract as the native tests, run against real silicon, using the shared driver-check helpers from #121 to guarantee the two sides exercise the same logic.

## Goal

A hardware-unit suite at `tests/hardware/test_hts221/test_main.cpp` that:
- Talks to a real HTS221 on the STeaMi internal I2C bus.
- Uses `tests/shared/driver_checks.h` helpers so the **same** checks the native suite validates with mock-exact values are re-validated on silicon with plausibility ranges.
- Serves as the template every future driver's hardware-unit suite copies.

## Acceptance criteria

- [ ] New file `tests/hardware/test_hts221/test_main.cpp`:
  - SPDX header, `#include <HTS221.h>`, `#include "driver_checks.h"`.
  - Instantiates `TwoWire internalI2C(I2C_INT_SDA, I2C_INT_SCL)` and `HTS221 sensor(internalI2C)`.
  - Calls at minimum: `check_begin(sensor)`, `check_who_am_i(sensor, HTS221_WHO_AM_I_VALUE)`, `check_read_plausible(sensor, temp range, humidity range)`.
  - Temperature range: `0.0f .. 50.0f` °C (indoor-plus-margin). Humidity range: `10.0f .. 90.0f` %RH.
  - File header comment explains why those ranges were chosen.
- [ ] `make test-hardware-hts221` (from #120) passes on a STeaMi board plugged into the contributor's machine.
- [ ] The test suite is wired into the `make test-hts221` composite once #120 lands.

## Scope

~60 lines of C++, copy-paste-adjust from the native suite's new shape. Few hours once #121 is merged.

## Dependencies

- #121 (shared `driver_checks.h`) — this suite is the first user of it.
- #120 (per-driver hardware targets) — provides `make test-hardware-hts221`.

## Out of scope

- Integration-level checks (long-running streaming, BDU tearing) — those live in the #7 integration issue.
- Qualification (human-in-the-loop "does breathing on it raise humidity?") — see qualification issue.

## Related

Sibling issues: integration #TBD and qualification #TBD provide levels 2 and 3 for HTS221.

### PR DESCRIPTION
**Pull request by Charly-sketch**
## Summary

Add the first level-1 hardware-unit validation suite for the HTS221 driver, running on a real STeaMi board and reusing the shared `driver_checks.h` helpers introduced for native/hardware parity.

Closes #122

## Changes

* Add `tests/hardware/test_hts221/test_main.cpp`
* Instantiate `TwoWire internalI2C(I2C_INT_SDA, I2C_INT_SCL)` for the onboard internal I2C bus
* Instantiate `HTS221 sensor(internalI2C)` on real hardware
* Add `check_begin(sensor)` validation
* Add `check_who_am_i(sensor, HTS221_WHO_AM_I_VALUE)` validation
* Add plausible silicon temperature validation (`0.0f` → `50.0f`)
* Add plausible silicon humidity validation (`10.0f` → `90.0f`)
* Run the suite through Unity so each parity check is reported independently
* Establish the reference hardware-unit template that future onboard sensor drivers can copy

## Checklist

* [x] `make lint` passes (clang-format)
* [x] `make build` passes (PlatformIO)
* [x] `make test-native` passes (if native tests exist)
* [x] `make test-hardware` passes on a connected STeaMi (if hardware tests exist)
* [x] Commit messages follow conventional commits format

### CODE REVIEW

**Comment by nedseb** [comment]
J'ai rebasé sur main par cherry-pick de ton commit `1767a38` — les 6 commits précédents étaient ceux des PRs #137, #144 et #148, toutes mergées maintenant. Aucune modification de contenu de ma part : le sketch hardware-unit pour HTS221 part tel quel.

J'ai aussi retargeté la PR de `test/shared-driver-checks` (branche supprimée après merge de #148) vers `main`. MERGEABLE confirmé.

**Validé sur la STeaMi :**

```
$ make test-hardware-hts221
...
tests/hardware/test_hts221/test_main.cpp:44: test_hts221_begin                       [PASSED]
tests/hardware/test_hts221/test_main.cpp:45: test_hts221_who_am_i                    [PASSED]
tests/hardware/test_hts221/test_main.cpp:46: test_hts221_read_plausible_temperature  [PASSED]
tests/hardware/test_hts221/test_main.cpp:47: test_hts221_read_plausible_humidity     [PASSED]
4 test cases: 4 succeeded in 00:00:11.675
```

Bonus de cohérence avec ce qui est sur main : l'auto-découverte par foreach+eval livrée dans #144 picke ton nouveau dossier `tests/hardware/test_hts221/` automatiquement, donc `make test-hardware-hts221` est déjà disponible (et `make test-hardware` l'inclut aussi). Pas de Makefile à toucher.

Attribution Charly conservée comme **author** du commit (pas juste co-author trailer) grâce au cherry-pick. Prête à merger.

File: tests/hardware/test_hts221/test_main.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    check_who_am_i(sensor, HTS221_WHO_AM_I_VALUE);
+}
+
+void test_hts221_read_plausible_temperature() {
+    check_read_plausible(sensor, &HTS221::temperature, 0.0f, 50.0f);
+}
+
+void test_hts221_read_plausible_humidity() {
+    check_read_plausible(sensor, &HTS221::humidity, 10.0f, 90.0f);
+}
```
[the remark this answers is not in the document]
Comment (nedseb) [reply tests/hardware/test_hts221/test_main.cpp:55]: Bien vu, fix poussé en `afaaa2d`. Ajout d'un `setUp()` qui appelle `sensor.begin()` avant chaque test pour rendre les tests indépendants.

Le fix a été un poil plus large que l'analyse initiale : `begin()` se termine par `powerOff()`, donc avec juste `begin()` dans `setUp()`, `humidity()` partait sur le chemin auto-trigger one-shot — qui sur silicium réel renvoyait parfois `0.0f` (raw négatif clamped par `computeHumidity`). J'ai aligné le `setUp()` sur la recette des tests natifs de plausibilité : `begin()` + `setContinuous(HTS221_ODR_1_HZ)` pour rester en mode continu et éviter ce comportement.

L'edge case dans le driver lui-même (auto-trigger one-shot après `begin()` qui produit occasionnellement un raw négatif côté humidité) est tracké séparément dans #157 — la dépendance du test au `setContinuous` deviendra optionnelle une fois le driver corrigé.

4/4 tests passent post-fix sur la STeaMi physique.
